src/cargo.rs

```diff
@@ -0,0 +1,140 @@
+use std::env;
+use std::fs;
+use std::path::PathBuf;
+use std::io::prelude::*;
+use err::ProfError;
+use regex::Regex;
+use std::process::{Command, exit};
+use std::path::Path;
+
+/// Returns the closest ancestor path containing a `Cargo.toml`.
+///
+/// Returns `None` if no ancestor path contains a `Cargo.toml`, or if
+/// the limit of MAX_ANCESTORS ancestors has been reached.
+pub fn find_toml() -> Option<PathBuf> {
```

Comment:
agh, was trying to find a command like that! thanks for pointing it out, makes my life easier.

i'll change the get_package_name/find_toml functions to go through `cargo read-manifest`. i think we'll still need the find_target, unless i'm wrong